src/main/java/study/step4/model/car/Cars.java

```diff
@@ -0,0 +1,56 @@
+package study.step4.model.car;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Stream;
+import study.step4.model.strategy.RandomMoveStrategy;
+
+public class Cars {
+
+  private final List<Car> racingGameCars;
+
+  private Cars(String[] carNames) {
+    this.racingGameCars = new ArrayList<>();
+    for (String car : carNames) {
+      this.racingGameCars.add(new Car(car));
+    }
+  }
+
+  private Cars(List<Car> cars) {
+    this.racingGameCars = cars;
+  }
+
+  public static Cars of(List<Car> cars) {
+    return new Cars(cars);
+  }
+
+  public static Cars of(String[] carNames) {
+    return new Cars(carNames);
+  }
+
+  public static Cars of(int carNumber) {
+    return Stream.generate(Car::new)
+        .limit(carNumber)
+        .collect(collectingAndThen(toList(), Cars::of));
+  }
```

Comment:
```suggestion
  private Cars(List<Car> cars) {
    this.racingGameCars= cars;
  }

  public static Cars of(List<Car> cars) {
    return new Cars(cars);
  }

  public static Cars of(String[] carNames) {
    return Arrays.stream(carNames)
            .map(Car::new)
            .collect(collectingAndThen(toList(), Cars::of));
  }
  
  public static Cars of(String carNames) {
    return of(carNames.split(",")); // String도 input으로 받을 수 있게 해주세요.
  }
  
  public static Cars of(int carNumber) {
    return Stream.generate(Car::of)
        .limit(carNumber)
        .collect(collectingAndThen(toList(), Cars::of));
  }
```

이렇게 표현해주면 어떨까요? 불필요하게 생성자가 두개일 필요는 없을 것 같아요.